crates/sui-indexer/migrations/pg/2023-10-06-204335_tx_indices/up.sql

```diff
@@ -54,14 +62,24 @@ CREATE TABLE tx_calls_fun (
     PRIMARY KEY(package, module, func, tx_sequence_number)
 );
 CREATE INDEX tx_calls_fun_sender ON tx_calls_fun (sender, package, module, func, tx_sequence_number);
+CREATE INDEX tx_calls_fun_tx_sequence_number ON tx_calls_fun (tx_sequence_number);
 
 CREATE TABLE tx_digests (
     tx_digest                   BYTEA        PRIMARY KEY,
     tx_sequence_number          BIGINT       NOT NULL
 );
+CREATE INDEX tx_digests_tx_sequence_number ON tx_digests (tx_sequence_number);
 
 CREATE TABLE tx_kinds (
     tx_sequence_number          BIGINT       NOT NULL,
     tx_kind                     SMALLINT     NOT NULL,
     PRIMARY KEY(tx_kind, tx_sequence_number)
 );
+CREATE INDEX tx_kinds_tx_sequence_number ON tx_kinds (tx_sequence_number);
+
+CREATE TABLE cp_tx (
```

Comment:
@wlmyng I saw you used to have this table, did we decide to remove it? it's useful for tx based pruning here.

Reply:
I ended up adding this information onto the `checkpoints` table. Although more recently I figured out you can figure the tx sequence number range of a checkpoint using network_total_transactions - the current checkpoint's first tx sequence number is the previous checkpoint's network_total_transactions, and by the time we index a checkpoint, it's network_total_transactions - 1 is its last tx sequence number

Reply:
got it, thanks for the context. however with the pruning logic in, when we try to prune checkpoint X, checkpoint X-1 is gone already, so deriving that would not work here.
also I would like to split again, so that
- we will delete rows from checkpoints table first as we discussed on last meeting, so that from GQL side, we can count on this table for "available range"
- and delete rows from cp_tx table last, as the watermark table, and to handle disaster recovery too

Reply:
synced offline with will, will add back the cp_tx table